React/Views/RCTScrollView.m

```diff
@@ -116,6 +116,29 @@ - (BOOL)canCoalesce
 
 - (id<RCTEvent>)coalesceWithEvent:(id<RCTEvent>)newEvent
 {
+  NSMutableArray *updatedChildFrames = [[NSMutableArray alloc] init];
+
+  NSMutableArray *oldUpdatedChildFrames = [self.body objectForKey:@"updatedChildFrames"];
```

Comment:
You can just change `newEvent` from `id<RCTEvent>` to `RCTScrollEvent *` and something shorter, like:

```objc
NSArray *updatedChildFrames = [_userData[@"updatedChildFrames"] arrayByAddingObjectsFromArray:newEvent->_userData[@"updatedChildFrames"]];

if (updatedChildFrames) {
  NSMutableDictionary *userData = [newEvent->_userData mutableCopy];
  userData[@"updatedChildFrames"] = updatedChildFrames;
  newEvent->_userData = userData;
}

return newEvent;
```

Also, I changed so it doesn't replace the whole `userData`, so in case more information is added, it doesn't get erased.